src/setups/extrude_geometry.jl

```diff
@@ -31,45 +31,54 @@ Extrude either a line, a plane or a shape along a specific direction.
 
-# 3D Plane
+# Extrude a parallelogram in 3D to a parallelepiped in 3D
 p1 = [0.0, 0.0, 0.0]
 p2 = [0.5, 1.0, 0.0]
 p3 = [1.0, 0.2, 0.0]
 
 direction = [0.0, 0.0, 1.0]
 
-shape = ExtrudeGeometry((p1, p2, p3); direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+shape = extrude_geometry((p1, p2, p3); direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
 
-# Extrude a 2D shape to a 3D shape
+# Extrude a 2D shape (here: a disc) to a 3D shape (here: a cylinder)
 shape = SphereShape(0.1, 0.5, (0.2, 0.4), 1000.0, n_layers=3,
                     sphere_type=RoundSphere(end_angle=pi))
 
 direction = [0.0, 0.0, 1.0]
 
-shape = ExtrudeGeometry(shape; direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+shape = extrude_geometry(shape; direction, particle_spacing=0.1, n_extrude=4, density=1000.0)
+
+# output
+InitialCondition{Float64}(0.1, [0.44999999999999996 0.43096988312782164 … -0.23871756048182058 -0.24999999999999994; 0.4 0.4956708580912724 … 0.5001344202803415 0.4000000000000001; 0.05 0.05 … 0.35000000000000003 0.35000000000000003], [0.0 0.0 … 0.0 0.0; 0.0 0.0 … 0.0 0.0; 0.0 0.0 … 0.0 0.0], [1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002  …  1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002, 1.0000000000000002], [1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0  …  1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0, 1000.0], [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0  …  0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
 ```
 
-!!! warning
-    `particle_spacing` between extrusion layers may differ from shapes `particle_spacing`.
+!!! warning "Experimental Implementation"
+    This is an experimental feature and may change in any future releases.
 """
-function ExtrudeGeometry(geometry; particle_spacing, direction, n_extrude=1,
-                         velocity=zeros(length(direction)), tlsph=false,
-                         mass=nothing, density=nothing, pressure=0.0)
+function extrude_geometry(geometry; particle_spacing=-1, direction, n_extrude=2,
+                          velocity=zeros(length(direction)), tlsph=false,
+                          mass=nothing, density=nothing, pressure=0.0)
     direction_ = normalize(direction)
     NDIMS = length(direction_)
 
-    geometry = consider_particle_placement(geometry, direction_, particle_spacing, tlsph)
+    if !(geometry isa InitialCondition) && particle_spacing == -1
+        throw(ArgumentError("`particle_spacing` must be specified when not extruding an `InitialCondition`"))
+    elseif geometry isa InitialCondition
+        particle_spacing = geometry.particle_spacing
+    end
```

Comment:
```suggestion
    if geometry isa InitialCondition && geometry.particle_spacing > 0
        if particle_spacing > 0 && particle_spacing != geometry.particle_spacing
            throw(ArgumentError("`particle_spacing` must be -1 when using an `InitialCondition`"))
        end
        particle_spacing = geometry.particle_spacing
    end
    
    if particle_spacing <= 0
        throw(ArgumentError("`particle_spacing` must be specified when not extruding an `InitialCondition`"))
    end
```
The error message is not 100% correct, as we also print "`particle_spacing` must be specified when not extruding an `InitialCondition`" when an `InitialCondition` with `InitialCondition.particle_spacing == -1` is used, but I think explaining that in an error message is too complicated and confusing.